src/run.rs

```diff
@@ -36,27 +39,25 @@ pub fn run(
     let mut preview_area: Option<PreviewArea> = None;
 
     while context.quit == QuitType::DoNot {
-        backend.render(TuiView::new(context));
+        // do the ui
+        if let Ok(area) = backend.terminal_ref().size() {
+            // pre-calculate some ui attributes
+            calculate_ui_context(context, area);
```

Comment:
I introduced this `UiContext` struct which can be used in general to store all that UI attributes which are calculate before the render process to have it accessible also for the app logic.